### PR TITLE
Implement end points (backend) for a user to track/view his dossiers.

### PR DESCRIPTION
**Pull request by zihao429**

### CODE REVIEW

File: Server/ConcordiaCurriculumManager/Properties/launchSettings.json [hunk 1]
```diff
@@ -1,24 +1,33 @@
 {
+  "$schema": "https://json.schemastore.org/launchsettings.json",
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/Properties/launchSettings.json:2]: Was this file modified manually by you?

File: Server/ConcordiaCurriculumManager/Services/DossierService.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+namespace ConcordiaCurriculumManager.Services;
+
+public interface IDossierService
+{
+    public Task<List<Dossier>> GetDossiersByID(Guid ID);
+}
+
+public class DossierService : IDossierService
+{
+    private readonly ILogger<DossierService> _logger;
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/Services/DossierService.cs:18]: Can you remove all these unused class attributes?

File: Server/ConcordiaCurriculumManager/Controllers/DossierController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    [HttpGet(nameof(GetDossiersByID))]
+    [SwaggerResponse(StatusCodes.Status200OK, "Dossiers retrieved")]
+    [SwaggerResponse(StatusCodes.Status401Unauthorized, "User is not authorized")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    public async Task<ActionResult> GetDossiersByID() {
+        try
+        {
+            Guid userId = (await _userService.GetCurrentUser()).Id;
+            var dossiers = await _dossierService.GetDossiersByID(userId);
+            var dossiersDTOs = _mapper.Map<List<DossierDTO>>(dossiers);
```
Comment (JamesPartsafas) [inline Server/ConcordiaCurriculumManager/Controllers/DossierController.cs:46]: Before returning the data, can you log (with `_logger.LogInformation`) the data that is being returned?

**Comment by JamesPartsafas** [comment]
Everything looks good, but can you write some tests for this?

File: Server/ConcordiaCurriculumManager/Properties/launchSettings.json (hunk 1 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/Properties/launchSettings.json:2]: No, it was not

File: Server/ConcordiaCurriculumManager/Services/DossierService.cs (hunk 2 above)
Comment (zihao429) [reply Server/ConcordiaCurriculumManager/Services/DossierService.cs:18]: Sounds good

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**